internal/services/appservice/helpers/app_stack.go

```diff
@@ -0,0 +1,662 @@
+						"site_config.0.application_stack.0.java_version",
+					},
+					Description: "Should the application use the embedded web server for the version of Java in use.",
+				},
+
+				"tomcat_version": {
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					ValidateFunc: validation.StringIsNotEmpty, // This is a long list of regularly changing values, not all valid values of which are made known in the portal/docs
+					ConflictsWith: []string{
+						"site_config.0.application_stack.0.java_embedded_server_enabled",
+					},
+					RequiredWith: []string{
+						"site_config.0.application_stack.0.java_version",
+					},
+				},
+
+				"java_container": {
+					Type:       pluginsdk.TypeString,
+					Optional:   true,
+					Deprecated: "this property has been deprecated in favour of `tomcat_version` and `java_embedded_server_enabled`",
+					ValidateFunc: validation.StringInSlice([]string{
+						"JAVA",
+						"JETTY", // No longer supported / offered - Java SE or Tomcat (10, 9.5, 8) only
+						"TOMCAT",
+					}, false),
+					RequiredWith: []string{
+						"site_config.0.application_stack.0.java_container_version",
+					},
+					ConflictsWith: []string{
+						"site_config.0.application_stack.0.tomcat_version",
+					},
+				},
+
+				"java_container_version": {
+					Type:       pluginsdk.TypeString,
+					Optional:   true,
+					Deprecated: "This property has been deprecated in favour of `tomcat_version` and `java_embedded_server_enabled`",
+					RequiredWith: []string{
+						"site_config.0.application_stack.0.java_container",
+					},
+				},
+
+				"docker_container_name": {
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+					AtLeastOneOf: []string{
+						"site_config.0.application_stack.0.docker_container_name",
+						"site_config.0.application_stack.0.dotnet_version",
+						"site_config.0.application_stack.0.dotnet_core_version",
+						"site_config.0.application_stack.0.java_version",
+						"site_config.0.application_stack.0.node_version",
+						"site_config.0.application_stack.0.php_version",
+						"site_config.0.application_stack.0.python_version",
+						"site_config.0.application_stack.0.python",
+					},
+					RequiredWith: []string{
+						"site_config.0.application_stack.0.docker_container_tag",
+					},
+				},
+
+				"docker_container_registry": {
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+
+				"docker_container_tag": {
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+					RequiredWith: []string{
+						"site_config.0.application_stack.0.docker_container_name",
+					},
+				},
+
+				"current_stack": {
+					Type:     pluginsdk.TypeString,
+					Optional: true,
+					Computed: true, // This will be set to the configured type from above if not explicitly set
```

Comment:
Since windows can have multiple stacks "configured", and this simply controls which one the Portal displays, I feel it should remain for users with the edge case that may want to explicitly set which is presented?